prusa/connect/printer/models.py

```diff
@@ -155,6 +155,36 @@ def __repr__(self):
                 f" [{self.source}], {data}")
 
 
+class Snapshot(LoopObject):
+    """Snapshot from the camera"""
+
+    endpoint = "/c/snapshot"
+    method = "PUT"
+    needs_token = True
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, camera_fingerprint: str, camera_token: str, data: bytes,
+                 timestamp: float, fail_cb):
+        super().__init__(timestamp=timestamp)
+        self.camera_fingerprint = camera_fingerprint
+        self.camera_token = camera_token
+        self.data = data
+        self.fail_cb = fail_cb
+
+    def send(self, conn: Session, server, headers):
+        """A universal send function"""
+        name = self.__class__.__name__
+        log.debug("Sending %s: %s", name, self)
+        res = conn.request(method=self.method,
+                           url=server + self.endpoint,
+                           headers=headers,
+                           json=self.to_payload(),
```

Comment:
I told you, that camera data (image) is binary data, not json!

Reply:
My mistake, I forgot original function here